external/avro/src/main/scala/org/apache/spark/sql/avro/AvroUtils.scala

```diff
@@ -228,6 +242,40 @@ private[sql] object AvroUtils extends Logging {
       .groupBy(_.name.toLowerCase(Locale.ROOT))
       .mapValues(_.toSeq) // toSeq needed for scala 2.13
 
+    /** Get the fields which have matching equivalents in both Avro and Catalyst schemas. */
+    def getMatchedFields: Seq[AvroMatchedField] = catalystSchema.zipWithIndex.flatMap {
```

Comment:
let's store this in a `val`. My previous comment for "not storing" is for adding new code for validation only.

Reply:
Sure makes sense.